File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/Encrypted.java
package com.sequenceiq.cloudbreak.domain;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Indicates that a String field is encrypted, thus it needs to be decrypted before use.
 * The decryption happens on the converter side.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface Encrypted {
}

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/Encrypted.kt
package com.sequenceiq.cloudbreak.domain

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Indicates that a String field is encrypted, thus it needs to be decrypted before use.
 * The decryption happens on the converter side.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class Encrypted